File: connectors/java/src/main/java/com/mochilink/connector/subscription/EventSubscription.java
package com.mochilink.connector.subscription;

import java.util.List;
import java.util.Map;
import java.util.Objects;

/**
 * Represents an event subscription with filtering capabilities
 * 
 * Stores subscription information including event types, filters,
 * and metadata for managing event delivery.
 */
public class EventSubscription {
    
    private final String id;
    private final List<String> eventTypes;
    private final Map<String, Object> filters;
    private final long createdAt;
    
    public EventSubscription(String id, List<String> eventTypes, Map<String, Object> filters, long createdAt) {
        this.id = id;
        this.eventTypes = eventTypes;
        this.filters = filters;
        this.createdAt = createdAt;
    }
    
    /**
     * Check if this subscription includes the given event type
     */
    public boolean includesEventType(String eventType) {
        return eventTypes.contains(eventType);
    }
    
    /**
     * Check if event data matches the subscription filters
     */
    public boolean matchesFilters(Map<String, Object> eventData) {
        // No filters means accept all
        if (filters == null || filters.isEmpty()) {
            return true;
        }
        
        // Check each filter condition
        for (Map.Entry<String, Object> filter : filters.entrySet()) {
            String key = filter.getKey();
            Object expectedValue = filter.getValue();
            
            // Get actual value from event data
            Object actualValue = eventData.get(key);
            
            // If filter key doesn't exist in event data, filter doesn't match
            if (actualValue == null) {
                return false;
            }
            
            // Compare values
            if (!Objects.equals(actualValue, expectedValue)) {
                return false;
            }
        }
        
        return true;
    }
    
    // Getters
    public String getId() {
        return id;
    }
    
    public List<String> getEventTypes() {
        return eventTypes;
    }
    
    public Map<String, Object> getFilters() {
        return filters;
    }
    
    public long getCreatedAt() {
        return createdAt;
    }
    
    @Override
    public String toString() {
        return String.format("EventSubscription{id='%s', eventTypes=%s, filters=%s}", 
                           id, eventTypes, filters);
    }
}

File: mochi-link-connectors/bedrock-edition/pmmp/src/com/mochilink/connector/pmmp/subscription/EventSubscription.php
<?php

declare(strict_types=1);

namespace com\mochilink\connector\pmmp\subscription;

/**
 * Represents an event subscription with filtering capabilities
 * 
 * Stores subscription information including event types, filters,
 * and metadata for managing event delivery.
 * 
 * @author chm413
 * @version 1.0.0
 */
class EventSubscription {
    
    private string $id;
    private array $eventTypes;
    private array $filters;
    private int $createdAt;
    
    public function __construct(string $id, array $eventTypes, array $filters, int $createdAt) {
        $this->id = $id;
        $this->eventTypes = $eventTypes;
        $this->filters = $filters;
        $this->createdAt = $createdAt;
    }
    
    /**
     * Check if this subscription includes the given event type
     */
    public function includesEventType(string $eventType): bool {
        return in_array($eventType, $this->eventTypes, true);
    }
    
    /**
     * Check if event data matches the subscription filters
     */
    public function matchesFilters(array $eventData): bool {
        // No filters means accept all
        if (empty($this->filters)) {
            return true;
        }
        
        // Check each filter condition
        foreach ($this->filters as $key => $expectedValue) {
            // If filter key doesn't exist in event data, filter doesn't match
            if (!isset($eventData[$key])) {
                return false;
            }
            
            $actualValue = $eventData[$key];
            
            // Compare values
            if ($actualValue !== $expectedValue) {
                return false;
            }
        }
        
        return true;
    }
    
    // Getters
    public function getId(): string {
        return $this->id;
    }
    
    public function getEventTypes(): array {
        return $this->eventTypes;
    }
    
    public function getFilters(): array {
        return $this->filters;
    }
    
    public function getCreatedAt(): int {
        return $this->createdAt;
    }
    
    public function __toString(): string {
        return sprintf(
            "EventSubscription{id='%s', eventTypes=%s, filters=%s}",
            $this->id,
            json_encode($this->eventTypes),
            json_encode($this->filters)
        );
    }
}
